crates/biome_js_formatter/src/js/bindings/parameters.rs

```diff
@@ -26,8 +50,23 @@ impl FormatNodeRule<JsParameters> for FormatJsParameters {
     }
 }
 
-declare_node_union! {
-    pub(crate) FormatAnyJsParameters = JsParameters | JsConstructorParameters
+pub(crate) enum AnyJsParameters {
+    JsParameters(JsParameters),
+    JsConstructorParameters(JsConstructorParameters),
+}
+
+pub(crate) struct FormatAnyJsParameters {
+    pub(crate) parameters: AnyJsParameters,
+    pub(crate) options: FormatJsParametersOptions,
+}
+
+impl FormatAnyJsParameters {
+    pub(crate) fn new(parameters: AnyJsParameters, options: FormatJsParametersOptions) -> Self {
+        Self {
+            parameters,
+            options,
+        }
+    }
```

Comment:
I copied most of this from `FormatTemplateElement`, as another example of a formatting node that takes in options for a union of node types.